HPCC-23824 Provide HELM chart level logging configuration

rpastrana opened the pull request:
- Declares logging configuration structure
- Adds logging attributes to known hpcc components

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

Commit:
HPCC-23824 Provide HELM chart level logging configuration

- Declares logging configuration structure
- Adds logging attributes to known hpcc components

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

rpastrana commented:
@ghalliday please review

HPCCSmoketest commented:
Process of PR-13618, label: HPCC-23824-LogginConfig starts now.
The reason of this test is: Forced to re-test.
Commit ID: 79a5ae0962f87a7886ba2d1382bdc00b24757972
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-39.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-39.ca-central-1.compute.internal
Sha: 79a5ae0962f87a7886ba2d1382bdc00b24757972
Build: success
Milestone:Install hpccsystems-platform-community_7.8.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |54 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |102 sec  (00:01:42) |
|setup (roxie) |11 |11 |0 |18 sec  (00:00:18) |
|test (hthor) |873 |873 |0 |312 sec  (00:05:12) |
|test (thor) |812 |812 |0 |713 sec  (00:11:53) |
|test (roxie) |953 |953 |0 |380 sec  (00:06:20) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  657 sec (00:10:57) |  119 sec (00:01:59) |   22 sec (00:00:22) |   18 sec (00:00:18) | 1792 sec (00:29:52) |   20 sec (00:00:20) | 2637 sec (00:43:57) |

HPCCSmoketest commented:
Process of PR-13618, label: HPCC-23824-LogginConfig starts now.
The reason of this test is: Did not build yet.
Commit ID: 79a5ae0962f87a7886ba2d1382bdc00b24757972
Estimated completion time is ~0.9 hour(s)
centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: 79a5ae0962f87a7886ba2d1382bdc00b24757972
Build: success
Milestone:Install hpccsystems-platform-community_7.8.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |55 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |47 sec  (00:00:47) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |873 |873 |0 |579 sec  (00:09:39) |
|test (thor) |812 |812 |0 |976 sec  (00:16:16) |
|test (roxie) |953 |953 |0 |704 sec  (00:11:44) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   17 sec (00:00:17) |  925 sec (00:15:25) |  123 sec (00:02:03) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2596 sec (00:43:16) |   17 sec (00:00:17) | 3718 sec (01:01:58) |

ghalliday commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -246,6 +261,44 @@
+      "type": "object",
+      "properties": {
+        "fields": {
+          "type": "string",
```
Would this be more natural in a HELM file as a list rather than a string?  Same for audiences/classes.  (I don't know the answer.)

rpastrana replied:
Yes, we discussed making these a list, it was decided instead to carry over the legacy format which utilizes pre-stablished logic,and is more flexible, allowing users to build up fields(with the + delimter) or build down with the - delimiter

ghalliday commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -246,6 +261,44 @@
+        },
+        "detail": {
+          "type": "integer",
+		  "description": "Log output verbosity"
```
indent: tabs instead of spaces have been used - other instances as well.

ghalliday commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -246,6 +261,44 @@
+        },
+		"queueLen": {
+          "type": "integer",
+		  "description": "Ignored if 'useLogQueue' is false"
```
This doesn't tell you what the property means

rpastrana replied:
Once I get a description of the field's function, I could fill in the description. Same for below field.

ghalliday commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -246,6 +261,44 @@
+        },
+		"queueDrop": {
+          "type": "integer",
+		  "description": "Ignored if 'useLogQueue' is false"
```
This doesn't describe the property

ghalliday reviewed (commented):
A few comments, but @richardkchapman would be much better at reviewing

richardkchapman commented:
My main concern really is wondering why we want to make these things configurable. Lots of configuration options means lots of things to document, lots of codepaths to test, lots of noise in the helm chart - and for what benefit?

rpastrana commented:
@richardkchapman should this be removed? Are you envisioning a pre-determined log output format for all components?

jakesmith commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -125,6 +125,9 @@
           "$ref": "#/definitions/storage"
+        },
+        "logging": {
+          "type": "object"
```
I think this needs to be a reference like the others, i.e.:
```
"logging": {
   "$ref": "#/definitions/logging"
}
```

Commit:
HPCC-23824 Containerized logging configuration changes

- Removes several fields deemed not worth of configuration
- Removes tabs
- Adds logging structure references

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

rpastrana commented:
@richardkchapman @ghalliday @jakesmith please review latest commit. 
Kept the logging options "classes", "audiences" and "detail", because I considered them to be the most sensible for users to configure based on their needs. however, if this doesn't fit our new configuration philosophy I can remove

HPCCSmoketest commented:
Process of PR-13618, label: HPCC-23824-LogginConfig starts now.
The reason of this test is: Forced to re-test.
Commit ID: e447d12726df2582d5779393aa5257ae7a3f87a0
Estimated completion time is ~0.5 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-53.ca-central-1.compute.internal

HPCCSmoketest commented:
Process of PR-13618, label: HPCC-23824-LogginConfig starts now.
The reason of this test is: New commit.
Commit ID: e447d12726df2582d5779393aa5257ae7a3f87a0
Estimated completion time is ~0.9 hour(s)
centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-53.ca-central-1.compute.internal
Sha: e447d12726df2582d5779393aa5257ae7a3f87a0
Build: success
Milestone:Install hpccsystems-platform-community_7.8.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |55 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |8 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |28 sec  (00:00:28) |
|setup (thor) |11 |11 |0 |43 sec  (00:00:43) |
|setup (roxie) |11 |11 |0 |21 sec  (00:00:21) |
|test (hthor) |873 |873 |0 |368 sec  (00:06:08) |
|test (thor) |812 |812 |0 |815 sec  (00:13:35) |
|test (roxie) |953 |953 |0 |396 sec  (00:06:36) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  660 sec (00:11:00) |  126 sec (00:02:06) |   22 sec (00:00:22) |   20 sec (00:00:20) | 1909 sec (00:31:49) |   18 sec (00:00:18) | 2764 sec (00:46:04) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: e447d12726df2582d5779393aa5257ae7a3f87a0
Build: success
Milestone:Install hpccsystems-platform-community_7.8.5-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |120 |120 |0 |0 |0 |53 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |29 sec  (00:00:29) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |873 |873 |0 |632 sec  (00:10:32) |
|test (thor) |812 |812 |0 |1019 sec  (00:16:59) |
|test (roxie) |953 |953 |0 |715 sec  (00:11:55) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  921 sec (00:15:21) |  123 sec (00:02:03) |   23 sec (00:00:23) |   17 sec (00:00:17) | 2722 sec (00:45:22) |   17 sec (00:00:17) | 3832 sec (01:03:52) |

jakesmith commented on dockerfiles/hpcc/values.schema.json:
```diff
@@ -246,8 +261,29 @@
+          "description": "List of target logging classes to include in output. Comprised of 3 letter codes delimited by + or -"
+        },
+        "detail": {
+          "type": "integer",
```
minor: could add a:
```
"minimum": 0 
```

.. to prevent invalid negative numbers.

jakesmith commented:
I think I can potentially see "classes" and "audience" being used by admins/devs. e.g. prod systems may want to tune down the logging way down to just warnings/errors..

@ghalliday @richardkchapman

ghalliday commented:
@rpastrana this needs rebasing and squashing.
then assign to @richardkchapman to review/merge

rpastrana commented:
@richardkchapman I thought this commit got usurped by one of your PRs and we were going to close this one.
